test/test_ops.py

```diff
@@ -334,12 +337,62 @@ def test_out(self, device, dtype, op):
         sample = samples[0]
         # call it normally to get the expected result
         expected = op(*sample.input, *sample.args, **sample.kwargs)
+        return expected, sample
+
+    @ops(op_db)
+    def test_out(self, device, dtype, op):
+        # Compute expected output of op
+        expected, sample = self.compute_expected_output(device, dtype, op)
         # call it with out=... and check we get the expected result
         out_kwargs = sample.kwargs.copy()
         out_kwargs['out'] = out = torch.empty_like(expected)
         op(*sample.input, *sample.args, **out_kwargs)
-        self.assertEqual(expected, out)
+        self.assertEqual(expected, out, exact_device=True)
+
+    @ops(op_db)
+    def test_out_empty_tensor(self, device, dtype, op):
+        # Compute expected output of op
+        expected, sample = self.compute_expected_output(device, dtype, op)
+        # call it with out=... and check we get the expected result
+        out_kwargs = sample.kwargs.copy()
+        out_kwargs['out'] = out = torch.empty(0,0, dtype=expected.dtype, device=expected.device)
+        op(*sample.input, *sample.args, **out_kwargs)
```

Comment:
This is great. Just wrap this to verify it doesn't throw a warning:

```
with warnings.catch_warnings(record=True) as w:
  warnings.simplefilter("always")
  op(*sample.input, *sample.args, **out_kwargs)
  self.assertEqual(len(w), 0)
```

Reply:
Added